Fixes for gallery height computation

rfrenchseti opened the pull request:
This doesn't fix any specific bugs.

rfrenchseti committed:
Fixes for gallery height computations. Move sliderViewableFraction to o_browse and rename to galleryImageViewableFraction. Calculate three Y heights - floor, ceil, partial. Rename tr to trFloor.

juzen2003 committed:
Fix the issue of startObs changes to the max slider value when pasting an
URL with startObs that is one row ahead of max slider value.
- In browse.js (line 723, 745, 771, 851, 1081), add "isScrolling" as a
  parameter to updateSliderHandle. It's used as one condition to determine
  if we should update the startObs to max slider value when ps reaches to
  the end (We do so when manually scrolling ps to the end). Note: every
  time when we first render initial data from renderGalleryAndTable, an
  event of ps reaching to the end will get triggered, in this case, we don't
  want to update the startObs to max slider value.

rfrenchseti committed:
Merge pull request #1 from juzen2003/rfrench_fix_max_slider

Fix the issue of startObs changes to the max slider value when pasting URL

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -2130,13 +2135,14 @@ var o_browse = {
         if ($(`${tab} .op-data-table tbody tr[data-obs]`).length > 0) {
-            trCount = o_utils.floor((height-$("th").outerHeight())/$(`${tab} .op-data-table tbody tr[data-obs]`).outerHeight());
+            trCountFloor = o_utils.floor((height-$("th").outerHeight()) /
+                                    $(`${tab} .op-data-table tbody tr[data-obs]`).outerHeight());
```
Please make this line align with the previous line after "="

rfrenchseti replied:
Done

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -2145,21 +2151,30 @@ var o_browse = {
         if (!o_browse.isGalleryView(view) && $(`${tab} .op-data-table tbody tr[data-obs]`).length > 0) {
-            trCount = o_utils.floor((height-$("th").outerHeight())/$(`${tab} .op-data-table tbody tr[data-obs]`).outerHeight());
+            trCountFloor = o_utils.floor((height-$("th").outerHeight()) /
+                                         $(`${tab} .op-data-table tbody tr[data-obs]`).outerHeight());
```
Same for this line.

rfrenchseti replied:
Done

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -2145,21 +2150,30 @@ var o_browse = {
+        // This is the total number of rows that are visible as much as we think is visually useful
+        let yCountPartial = Math.floor((height + o_browse.imageSize*(1-o_browse.galleryImageViewableFraction)) /
+                                       o_browse.imageSize);
         // let yCount = Math.ceil(height/o_browse.imageSize);
```
Do we want to use floor and ceil functions in utils.js? (because height could be a floating point value depending on window's height)

rfrenchseti replied:
Changed these and also a few other cases I found in browse.js

rfrenchseti committed:
Fix utils ceil function. Change new browse code to use utils functions and minor cosmetic cleanup.
